kernel-util/src/main/java/org/sakaiproject/content/util/ZipContentUtil.java

```diff
@@ -394,8 +401,36 @@ private void storeContentCollection(String rootId, ContentCollection collection,
 	 * @param out
 	 * @throws Exception
 	 */
-	private void storeContentResource(String rootId, ContentResource resource, ZipOutputStream out) throws Exception {		
-		String filename = resource.getId().substring(rootId.length(),resource.getId().length());				
+	private void storeContentResource(String rootId, ContentResource resource, ZipOutputStream out) throws Exception {
+		String filename = resource.getId().substring(rootId.length(),resource.getId().length());
+
+		//Inorder to have username as the folder name rather than having eids
+		if(filename != null && filename.length()>0) {
+			String filenameArr[] = filename.split(Entity.SEPARATOR);
+			if (filenameArr.length > 1) {
+				ContentCollectionEdit collectionEdit = (ContentCollectionEdit) ContentHostingService.getCollection(rootId+filenameArr[0]+Entity.SEPARATOR);
```

Comment:
You shouldn't have all this duplicated code. (https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)